src/MigrationTools.Clients.AzureDevops.ObjectModel/Processors/TfsPipelineProcessor.cs

```diff
@@ -70,267 +66,182 @@ private void MigratePipelines()
-                }
-            }
-            Log.LogInformation("From {sourcePipelines} source Pipelines {pipelinesToBeMigrated} Pipelines are going to be migrated..", sourceBuildDefinition.Count, pipelinesToBeMigrated.Count);
+            //Filter out Pipelines that already exist
+            var pipelinesToBeMigrated = sourceDefinitions.Where(s => !targetDefinitions.Any(t => t.Name == s.Name));
+
+            Log.LogInformation($"From {sourceDefinitions.Count} source {pipelineType} Pipelines {pipelinesToBeMigrated.Count()} Pipelines are going to be migrated..");
             string baseUrl = Target.Organisation + "/" + Target.Project + "/_apis/build/definitions?api-version=5.1-preview";
-            string credentials = Convert.ToBase64String(Encoding.ASCII.GetBytes(":" + Target.AccessToken));
 
-            foreach (BuildDefinition pipelineToBeMigrated in pipelinesToBeMigrated)
+            if (pipelineType == PipelineType.build)
             {
-                WebClient client = new WebClient();
-                client.Headers[HttpRequestHeader.Authorization] = "Basic " + credentials;
-                client.Headers[HttpRequestHeader.ContentType] = "application/json";
-                client.Headers.Add("user-agent", "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.2; .NET CLR 1.0.3705;)");
-
-                pipelineToBeMigrated.Links = null;
-                pipelineToBeMigrated.AuthoredBy = null;
-                pipelineToBeMigrated.Queue = null;
-                pipelineToBeMigrated.Url = null;
-                pipelineToBeMigrated.Uri = null;
-                pipelineToBeMigrated.Revision = 0;
-                pipelineToBeMigrated.Id = 0;
-                pipelineToBeMigrated.Project = null;
-                pipelineToBeMigrated.Repository.Id = null;
-
-                Log.LogInformation("Processing Pipeline '{pipelineToBeMigrated}'..", pipelineToBeMigrated.Name);
-                string body = JsonConvert.SerializeObject(pipelineToBeMigrated);
-                try
+                foreach (BuildDefinition pipelineToBeMigrated in pipelinesToBeMigrated)
                 {
-                    client.UploadString(baseUrl, "POST", body);
-                }
-                catch
-                {
-                    Log.LogError("Error migrating Pipeling '{pipelineToBeMigrated}'. Please migrate it manually.", pipelineToBeMigrated.Name);
-                }
-            }
-        }
+                    var client = getHttpClient(Target.AccessToken);
 
-        private void CreateReleaseDefinitions()
-        {
-            List<ReleaseDefinition> pipelinesToBeMigrated = new List<ReleaseDefinition>();
-            Log.LogInformation("Fetching Release Pipelines...");
-            GetReleaseDefinitions(Source.Organisation, Source.Project, Source.AccessToken);
-            GetReleaseDefinitions(Target.Organisation, Target.Project, Target.AccessToken);
+                    pipelineToBeMigrated.Links = null;
```

Comment:
this seems duplicated with the below code? why not have more in the baseclass and do this on the base type and have a generic method here that handles both cases like my previous comment? only here forcing the T : PipelineBase